core/src/main/java/hudson/model/ParametersDefinitionProperty.java

```diff
@@ -66,15 +66,30 @@
 public class ParametersDefinitionProperty extends OptionalJobProperty<Job<?, ?>>
         implements Action {
 
-    private final List<ParameterDefinition> parameterDefinitions;
+    private List<ParameterDefinition> parameterDefinitions;
 
     @DataBoundConstructor
-    public ParametersDefinitionProperty(List<ParameterDefinition> parameterDefinitions) {
-        this.parameterDefinitions = parameterDefinitions;
+    public ParametersDefinitionProperty(@Nonnull List<ParameterDefinition> parameterDefinitions) {
+        if (parameterDefinitions != null) {
+            this.parameterDefinitions = parameterDefinitions;
+        } else {
+            this.parameterDefinitions = Collections.emptyList();
+        }
     }
 
-    public ParametersDefinitionProperty(ParameterDefinition... parameterDefinitions) {
-        this.parameterDefinitions = Arrays.asList(parameterDefinitions);
+    public ParametersDefinitionProperty(@Nonnull ParameterDefinition... parameterDefinitions) {
+        if (parameterDefinitions != null) {
```

Comment:
DRY
use <code>this(parameterDefinitions != null ? Arrays.asList(parameterDefinitions) : null)</code>